### ISSUE TITLE
Tests failures on CI due to parallel testing

### ISSUE THREAD

**Issue by seisman** [GenericMappingTools/gmt#939]
We now have only 2 or 3 failing tests locally (on my mac). However, there are always more failing tests on CI.

The failing tests output error messages like:

```
373/839 Test #376: grdfill/nnfill.sh .......................***Failed    0.72 sec
ERROR 4: `/Users/vsts/.gmt/server/earth_relief_05m.grd' not recognized as a supported file format.
grdmath (api_import_grid): Not a supported grid format [earth_relief_05m.grd]
[Session gmt (0)]: Error returned from GMT API: GMT_GRID_READ_ERROR (18)
[Session gmt (0)]: Error returned from GMT API: GMT_GRID_READ_ERROR (18)
grdfill [ERROR]: Error for input file: No such file (islands.nc)
grdfill [ERROR]: Syntax error: Cannot find input file islands.nc
grdfill [ERROR]: Syntax error: Must specify input grid file
ERROR: gmtest:54
memtrack errors: 0
exit status: 1
```
The reason is, we run the tests in parallel using 4 jobs. Some jobs may download the same remote files (e.g. earth_relief_05m.grd) simultaneously, resulting in a corrupted local file.

Possible solutions:

1. Run tests using one job, but it will increase tests time on CI.
2. Run `gmt which` or similar to fetch these remote files before testing.

**Comment by PaulWessel** [comment]
What if those scripts using @eart_relief_xxx instead started with a
`gmt which -Gl @eart_relief_xxx
`
so they get a copy in the local directory.  Then the rest of the script leaves off the @.

**Comment by seisman** [comment]
Not only @eart_relief_xxx. Almost all remote files (the cache ones) have the same problem, as long as they're used by multiple scripts. And we also don't know the order in which these scripts are run.

**Comment by PaulWessel** [comment]
Yes, but most of those are small files that could be duplicated and given
separate names, and most are only used in one script.  And since they are
not in the repo (well, a separate repo for now) and just are downloaded it
should be fine.

More longterm I think we need to reorganize the tests suite anyway - this
is something we need to discuss at Scripps.  Since the names of the test
scripts do not tell what they do anyways (test//psxy/bugger.sh) we could go
with the system in seis an geodesy (seis_01.sh,seis_02.sh) and then when
needed could have seis_07.txt in cache and use @seis_07.txt.  It would be
easier to find which scripts uses what remote data.

So perhaps we dont need to fix this before 6beta.

Paul Wessel, Professor and Chair
Dept. of Earth Sciences (formerly Geology & Geophysics)
SOEST, U of Hawaii at Manoa


On June 15, 2019 at 7:26:06 AM, Dongdong Tian (notifications@github.com)
wrote:

Not only @eart_relief_xxx. Almost all remote files (the cache ones) have
the same problem, as long as they're used by multiple scripts. And we also
don't know the order in which these scripts are run.

—
You are receiving this because you commented.
Reply to this email directly, view it on GitHub
<https://github.com/GenericMappingTools/gmt/issues/939?email_source=notifications&email_token=AGJ7IX2J27O5YXO5IH6XZVDP2UQ25A5CNFSM4HYPNYH2YY3PNVWWK3TUL52HS4DFVREXG43VMVBW63LNMVXHJKTDN5WW2ZLOORPWSZGODXY4P5A#issuecomment-502384628>,
or mute the thread

**Comment by seisman** [comment]
Another question, I remember we have a md5sum file for @earth_relief_xx and it should be somewhere in the remote server. However, I cannot find the md5sum file in the ftp ftp://ftp.soest.hawaii.edu/gmt. Where is it? 

http://oceania.generic-mapping-tools.org is redirected to this site http://www.soest.hawaii.edu/gmt/data/, but I get a 403 forbidden error when visiting it.

Is http://www.soest.hawaii.edu/gmt/data/ the same as ftp://ftp.soest.hawaii.edu/gmt?

**Comment by PaulWessel** [comment]
No, the oceania URL redirects to http://www.soest.hawaii.edu/gmt/data, but there are no HTML there and not something to view.  In that sense it is hidden. @leouieda and I anticipate to add mirror servers on all continents, hence the name oceania for us in the Pacific.  This is all work in progress and perhaps we will move this to ftp at some point, but it should be transparent to the users.  The whole management of the md5 stuff as well as the files and their upkeep is something to discuss at the summit.

**Comment by seisman** [comment]
Fixed in #947.
